app/services/roster_entry_sorter.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Sorts roster entries by:
+# - Linked + accepted
+# - Linked + not accepted
+# - Not linked
+
+class RosterEntrySorter
+  attr_reader :entries, :assignment
+
+  def initialize(roster_entries, assignment)
+    @entries = roster_entries
+    @assignment = assignment
+  end
+
+  def sort
+    users_with_repo = @assignment.repos.map(&:user_id)
+
+    @entries.sort_by do |entry|
```

Comment:
TIL, I actually didn't know you could set the return value of a block using next O.o